rudder_airflow_provider/hooks/rudderstack.py

```diff
@@ -302,3 +307,84 @@ def poll_profile_run(self, profile_id: str, run_id: str) -> Dict[str, Any]:
                     f"Polling for runId: {run_id} for profile: {profile_id} timed out"
                 )
             time.sleep(self.poll_interval)
+
+class RudderStackETLHook(BaseRudderStackHook):
+    """
+    RudderStackETLHook to interact with RudderStack ETL API.
+    :params connection_id: `Conn ID` of the Connection to be used to configure this hook.
+    :params request_retry_delay: Time (in seconds) to wait between each request retry.
+    :params request_timeout: Time (in seconds) after which the requests to RudderStack are declared timed out.
+    :params request_max_retries: The maximum number of times requests to the RudderStack API should be retried before failing.
+    """
+
+    def __init__(
+        self,
+        connection_id: str,
+        request_retry_delay: int = DEFAULT_RETRY_DELAY,
+        request_timeout: int = DEFAULT_REQUEST_TIMEOUT,
+        request_max_retries: int = DEFAULT_REQUEST_MAX_RETRIES,
+        poll_timeout: float = None,
+        poll_interval: float = DEFAULT_POLL_INTERVAL_SECONDS,
+    ) -> None:
+        super().__init__(
+            connection_id, request_retry_delay, request_timeout, request_max_retries
+        )
+        self.poll_timeout = poll_timeout
+        self.poll_interval = poll_interval
+
+    def start_sync(self, source_id: str) -> str:
+        """Triggers a sync run and returns runId if successful, else raises Failure.
+
+        Args:
+            source_id (str): ETL source ID
+        """
+        if not source_id:
+            raise AirflowException("source_id is required to start an ETL sync")
+        self.log.info(f"Triggering sync run for source id: {source_id}")
+        return self.make_request(
```

Comment:
what if there is no runId? Shoule we raise an error

Reply:
This gets handled in BaseRudderStackHook [make_request](https://github.com/rudderlabs/rudder-airflow-provider/blob/main/rudder_airflow_provider/hooks/rudderstack.py#L118) handling. Essentially we are trusting on right response behavior of API which is true for this API.